File: BAEKJOON/basic_algo_lecture/0x03/3273.js
/*

이중 반복문


*/

const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const n = input[0];
const numbers = input[1]
  .split(" ")
  .map((ele) => Number(ele))
  .sort((a, b) => a - b);
const x = input[2];
let answer = 0;
let start = 0;
let end = n - 1;
// console.log("numbers: ", numbers);

while (start != end) {
  if (numbers[start] + numbers[end] == x) {
    answer++;
    start++;
  } else if (numbers[start] + numbers[end] > x) {
    end--;
  } else {
    start++;
  }
}
console.log(answer);

File: BAEKJOON/basic_algo_lecture/0x03/3273.swift
// import Foundation


let n = Int(readLine()!)!
let numbers = readLine()!.split(separator: " ").map{Int($0)!}
let sum = Int(readLine()!)!

var tmp = Array(repeating: 0, count: 2000001)
var count = 0

numbers.sort()

for i in 0..<n {
	if numbers[i] > sum {
		break
	}
	else if tmp[sum - numbers[i]] != 0 {
		count += 1
	}
	else if tmp[numbers[i]] == 0 {
		tmp[numbers[i]] = 1;
	}
}

print(count)

/*

let n = Int(readLine()!)!
var numbers = readLine()!.split(separator: " ").map{Int($0)!}
let sum = Int(readLine()!)!

var count = 0
var start = 0
var end = n - 1

numbers.sort()

while start != end {
	if numbers[start] + numbers[end] == sum {
		count += 1
		start += 1
	}
	else if numbers[start] + numbers[end] > sum {
		end -= 1
	}
	else {
		start += 1
	}
}

print(count)
*/